src/helpers/query-daily-data.ts

```diff
@@ -0,0 +1,93 @@
+registerDailyDataProvider("FitbitActiveMinutes", fitbitTotalActiveMinutesDataProvider, simpleAvailabilityCheck("Fitbit", ["MinutesVeryActive", "MinutesFairlyActive", "MinutesLightlyActive"]));
+registerDailyDataProvider("FitbitLightlyActiveMinutes", fitbitLightlyActiveMinutesDataProvider, simpleAvailabilityCheck("Fitbit", ["MinutesLightlyActive"]));
+registerDailyDataProvider("FitbitFairlyActiveMinutes", fitbitFairlyActiveMinutesDataProvider, simpleAvailabilityCheck("Fitbit", ["MinutesFairlyActive"]));
+registerDailyDataProvider("FitbitVeryActiveMinutes", fitbitVeryActiveMinutesDataProvider, simpleAvailabilityCheck("Fitbit", ["MinutesVeryActive"]));
+registerDailyDataProvider("FitbitBreathingRate", fitbitVeryActiveMinutesDataProvider, simpleAvailabilityCheck("Fitbit", ["BreathingRate"]));
+registerDailyDataProvider("FitbitCaloriesBurned", fitbitCaloriesBurnedDataProvider, simpleAvailabilityCheck("Fitbit", ["Calories"]));
+registerDailyDataProvider("FitbitElevatedHeartRateMinutes", fitbitElevatedHeartRateMinutesDataProvider, simpleAvailabilityCheck("Fitbit", ["HeartRateZone"]));
+registerDailyDataProvider("FitbitFatBurnHeartRateMinutes", fitbitFatBurnMinutesDataProvider, simpleAvailabilityCheck("Fitbit", ["HeartRateZone"]));
+registerDailyDataProvider("FitbitCardioHeartRateMinutes", fitbitCardioMinutesDataProvider, simpleAvailabilityCheck("Fitbit", ["HeartRateZone"]));
+registerDailyDataProvider("FitbitPeakHeartRateMinutes", fitbitPeakMinutesDataProvider, simpleAvailabilityCheck("Fitbit", ["HeartRateZone"]));
+registerDailyDataProvider("FitbitFloors", fitbitFloorsDataProvider, simpleAvailabilityCheck("Fitbit", ["HeartRateZone"]));
+registerDailyDataProvider("FitbitHrv", fitbitHrvDataProvider, simpleAvailabilityCheck("Fitbit", ["Floors"]));
+registerDailyDataProvider("FitbitRestingHeartRate", fitbitRestingHeartRateDataProvider, simpleAvailabilityCheck("Fitbit", ["RestingHeartRate"]));
+registerDailyDataProvider("FitbitSleepMinutes", fitbitTotalSleepMinutesDataProvider, simpleAvailabilityCheck("Fitbit", ["SleepLevelRem", "SleepLevelLight", "SleepLevelDeep", "SleepLevelAsleep"]));
+registerDailyDataProvider("FitbitLightSleepMinutes", fitbitLightSleepMinutesDataProvider, simpleAvailabilityCheck("Fitbit", ["SleepLevelLight"]));
+registerDailyDataProvider("FitbitRemSleepMinutes", fitbitRemSleepMinutesDataProvider, simpleAvailabilityCheck("Fitbit", ["SleepLevelRem"]));
+registerDailyDataProvider("FitbitDeepSleepMinutes", fitbitDeepSleepMinutesDataProvider, simpleAvailabilityCheck("Fitbit", ["SleepLevelDeep"]));
+registerDailyDataProvider("FitbitSpO2", fitbitSpO2DataProvider, simpleAvailabilityCheck("Fitbit", ["SpO2"]));
+registerDailyDataProvider("FitbitSteps", fitbitStepsDataProvider, simpleAvailabilityCheck("Fitbit", ["Steps"]));
+registerDailyDataProvider("GoogleFitSteps", googleFitStepsDataProvider, simpleAvailabilityCheck("GoogleFit", ["Steps"]));
```

Comment:
All of the strings here are the "magic" strings that an end user needs to know in order to pass them to the `dailyDataType` parameter of the `DeviceDataMonthChart`.  If you were to just be looking at the storybook, though, is there any way to know what the valid options are?

Also, is there any reason that these can't be an enumeration to add a bit more type safety?

Reply:
Hmm, yeah, that caused issues with the dictionary because you can't easily have an enum | string as a dictionary key, but it does seem like something we could do on the method call at least.

The issue is I wanted to be able to extend this downstream...if there is a new device data type or new way to calculate something that comes up...OR if we want to add some study-specific thing (daily mood score etc.) I ideally do not want to *have* to add to MDHUI first

Reply:
Ah, I see - if someone had a specific way they wanted to handle data, they can currently register their own daily calculator with an arbitrary key and pass that same key to the `DeviceDataMonthChart`.  I certainly see the value in that, even if it feels less "typescript-y".  

So then really my main concern was people not knowing *where* to find these keys for the ones we're supporting out of the box here.  Maybe we could add them as an export in this module?  

```
export const DailyDataTypes = { 
  AppleHealthDistanceWalkingRunning: "AppleHealthDistanceWalkingRunning", 
  AppleHealthFlightsClimbed: "AppleHealthFlightsClimbed", 
  AppleHealthHeartRateRange: "AppleHealthHeartRateRange"...
  // etc
} as const;
```

Its a little verbose, but it at least sort of "publishes" the things we support?

Reply:
Yep, agree something like that seems necessary

Reply:
Typescript actually supports enums, so I added one for all the daily data types.